x-pack/plugins/infra/server/lib/adapters/nodes/lib/create_node_item.ts

```diff
@@ -4,70 +4,27 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import {
-  InfraContainer,
-  InfraHost,
-  InfraHostMetrics,
-  InfraPod,
-} from '../../../../../common/graphql/types';
-import { InfraBucket, InfraNode, InfraNodeRequestOptions, InfraNodeType } from '../adapter_types';
+import { InfraNode, InfraNodeMetric } from '../../../../../common/graphql/types';
+import { InfraBucket, InfraNodeRequestOptions } from '../adapter_types';
 
 // TODO: Break these function into seperate files and expand beyond just documnet count
 // In the code below it looks like overkill to split these three functions out
 // but in reality the create metrics functions will be different per node type.
-function createHostMetrics(
+function createNodeMetrics(
   options: InfraNodeRequestOptions,
   node: InfraBucket,
   bucket: InfraBucket
-): InfraHostMetrics {
-  return { count: bucket.doc_count };
-}
-
-function createPodMetrics(
-  options: InfraNodeRequestOptions,
-  node: InfraBucket,
-  bucket: InfraBucket
-): InfraHostMetrics {
-  return { count: bucket.doc_count };
-}
-
-function createContainerMetrics(
-  options: InfraNodeRequestOptions,
-  node: InfraBucket,
-  bucket: InfraBucket
-): InfraHostMetrics {
-  return { count: bucket.doc_count };
+): InfraNodeMetric[] {
+  return [{ id: '1', name: 'count', value: bucket.doc_count }];
```

Comment:
So this ignores anything passed in as the `metrics` array in the query and always returns the count. If we decide to leave it at that for the initial use case, we should make sure to not lose track of this.

Reply:
Created #21825 to track this for future work.